data/binding/convert_helper.go

```diff
@@ -1,12 +1,58 @@
 package binding
 
 import (
+	"fmt"
 	"strconv"
+	"strings"
 
 	"fyne.io/fyne/v2"
 	"fyne.io/fyne/v2/storage"
 )
 
+func stripFormatPrecision(in string) string {
+	// quick exit if certainly not float
+	if !strings.ContainsAny(in, "f") {
+		return in
+	}
+
+	start := -1
+	end := -1
+	runes := []rune(in)
+	for i, r := range runes {
```

Comment:
Shouldn't you just be able to use the string directly here and below? Ranging over the string will give you runes and the rune slice is converted to strings everywhere either way. I

Reply:
Hmm, maybe? I read recently a warning about ranging on string but I don’t remember what it was so was being cautious. We need `[]rune` anyway for the slicing so using the raw string is just re-converting therefor overhead.

Reply:
I think we came to the conclusion on Slack that `[]rune` is the right thing. Can this be unblocked @Jacalz?

Reply:
Oh yeah, sorry. I forgot about that part.